azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1421,58 +1422,56 @@ private void finalizeFlows(ExecutableFlow flow) {
-        String alertType = options.getFlowParameters().get("alert.type");
-        Alerter alerter = alerters.get(alertType);
-        if (alerter != null) {
+    if(alertUser) {
+      ExecutionOptions options = flow.getExecutionOptions();
+      // But we can definitely email them.
+      Alerter mailAlerter = alerters.get("email");
+      if (flow.getStatus() == Status.FAILED || flow.getStatus() == Status.KILLED) {
+        if (options.getFailureEmails() != null && !options.getFailureEmails().isEmpty()) {
           try {
-            alerter.alertOnError(flow);
+            mailAlerter.alertOnError(flow);
           } catch (Exception e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-            logger.error("Failed to alert by " + alertType);
+            logger.error(e);
           }
-        } else {
-          logger.error("Alerter type " + alertType
-              + " doesn't exist. Failed to alert.");
         }
-      }
-    } else {
-      if (options.getSuccessEmails() != null
-          && !options.getSuccessEmails().isEmpty()) {
-        try {
-
-          mailAlerter.alertOnSuccess(flow);
-        } catch (Exception e) {
-          logger.error(e);
+        if (options.getFlowParameters().containsKey("alert.type")) {
+          String alertType = options.getFlowParameters().get("alert.type");
+          Alerter alerter = alerters.get(alertType);
+          if (alerter != null) {
+            try {
+              alerter.alertOnError(flow);
+            } catch (Exception e) {
+              // TODO Auto-generated catch block
+              e.printStackTrace();
+              logger.error("Failed to alert by " + alertType);
+            }
+          } else {
+            logger.error("Alerter type " + alertType + " doesn't exist. Failed to alert.");
+          }
         }
-      }
-      if (options.getFlowParameters().containsKey("alert.type")) {
-        String alertType = options.getFlowParameters().get("alert.type");
-        Alerter alerter = alerters.get(alertType);
-        if (alerter != null) {
+      } else {
+        if (options.getSuccessEmails() != null && !options.getSuccessEmails().isEmpty()) {
           try {
-            alerter.alertOnSuccess(flow);
+
+            mailAlerter.alertOnSuccess(flow);
           } catch (Exception e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-            logger.error("Failed to alert by " + alertType);
+            logger.error(e);
+          }
+        }
+        if (options.getFlowParameters().containsKey("alert.type")) {
+          String alertType = options.getFlowParameters().get("alert.type");
+          Alerter alerter = alerters.get(alertType);
+          if (alerter != null) {
+            try {
+              alerter.alertOnSuccess(flow);
+            } catch (Exception e) {
+              // TODO Auto-generated catch block
+              e.printStackTrace();
```

Comment:
Could you make a separate pull request later to fix the logging here and others in this file?